js/gestionPresupuesto.js

```diff
@@ -1,18 +1,358 @@
+let idGasto = 0;
 
 
-function actualizarPresupuesto() {
-    // TODO
+function actualizarPresupuesto(presupuestoActualizado) { 
+
+    let presupuestoAuxiliar = presupuestoActualizado;
+
+    if (presupuestoAuxiliar >= 0) {   
+        presupuesto = presupuestoAuxiliar;    
+        } else {
+            presupuestoAuxiliar = -1;     
+        }
+
+    return presupuestoAuxiliar;               
 }
 
+
 function mostrarPresupuesto() {
-    // TODO
+    let x = presupuesto;
+    return `Tu presupuesto actual es de ${x} €`;
+}
+
+function CrearGasto(descripcion, valor, fecha, ...etiquetas) {
+    this.descripcion = descripcion;
+
+    if (valor >= 0) {
+        this.valor = valor;
+    } else {
+        this.valor = 0;
+    }
+
+    if (fecha)
+    {
+        this.fecha = Date.parse(fecha);  
+    }   
+    else
+    {
+        this.fecha = Date.now(fecha);  
+    }
+
+    if (etiquetas === null)
+    {
+        this.etiquetas = [];
+    }
+    else
+    {
+        this.etiquetas = etiquetas;
+    }
+    
+    this.mostrarGastoCompleto = function()
+    {
+        let fechaFormatoLocalizado = new Date(fecha);
+
+        let gasto = (`Gasto correspondiente a ${this.descripcion} con valor ${this.valor} €.\nFecha: ${fechaFormatoLocalizado.toLocaleString()}\nEtiquetas:\n- ${etiquetas.join ("\n- ")}\n`);
+
+        return gasto;
+    }
+
+    this.actualizarFecha = function(fecha)
+    {
+        if (fecha = Date.parse(fecha))
+        {
+            this.fecha = fecha;
+        }        
+    }
+
+    this.anyadirEtiquetas = function(...etiquetas)
+    {
+        for (let el of etiquetas)
+        {
+            if (!this.etiquetas.includes(el))
+            {
+                this.etiquetas.push(el);
+            }
+        }  
+    }
+
+    this.borrarEtiquetas = function(...etiquetas)
+    {
+        let indice;  
+        for (let el of etiquetas)
+        {
+            indice = this.etiquetas.indexOf(el); 
+
+            if (indice >=0)
+            {
+                this.etiquetas.splice(indice, 1);  
+            }          
+        }
+    }
+
+
+    this.mostrarGasto = function() {
+        let gasto = (`Gasto correspondiente a ${this.descripcion} con valor ${this.valor} €`);
+        return gasto;
+    }
+
+    this.actualizarDescripcion = function (descripcionActualizada) {
+        this.descripcion = descripcionActualizada;
+    }
+
+    this.actualizarValor = function (valorActualizado) {
+        if (valorActualizado > 0) {
+            this.valor = valorActualizado;
+        }
+    }
+
+    //obtenerPeriodoAgrupacion - Función de un parámetro que devolverá el período de agrupación correspondiente al parámetro periodo de la función y a la fecha del gasto. 
+    //Si el período a agrupar es dia, el período de agrupación tendrá el formato aaaa-mm-dd; si es mes, tendrá el formato aaaa-mm; y si es anyo, tendrá el formato aaaa. Ejemplos:
+    this.obtenerPeriodoAgrupacion = function (periodo)
+    {
+        let fechaAux = fecha;
```

Comment:
Aquí igual: tienes que poner `this.fecha`. Además, como `this.fecha` es un timestamp, antes tendrás que pasarla a cadena de texto.

Reply:
Previa conversión a objeto `Date`